tests/foreman/sys/test_fam.py

```diff
@@ -172,6 +175,106 @@ def create_fake_module(module_target_sat, module_name, module_classes):
     create_fake_module(module_target_sat, 'fakemodule', ['init'])
 
 
+FAM_IDM_TEST_PLAYBOOKS = ['activation_key', 'content_view', 'job_invocation']
+
+
+@pytest.fixture(scope='module')
+def idm_sat(satellite_factory, ad_data):
+    """Yields a Satellite enrolled into IDM."""
+    new_sat = satellite_factory()
+    new_sat.enable_satellite_ipv6_http_proxy()
+    new_sat.register_to_cdn()
+    ipa_host = IPAHost(new_sat)
+    ipa_host.enroll_idm_and_configure_external_auth()
+    new_sat.execute(f'echo {settings.ipa.password} | kinit {settings.ipa.user}')
+    new_sat.cli.AuthLogin.negotiate()
+    yield new_sat
+    new_sat.unregister()
+    new_sat.teardown()
+    Broker(hosts=[new_sat]).checkin()
+
+
+@pytest.fixture(scope='module')
+def setup_fam_with_idm(idm_sat, module_sca_manifest):
+    # Execute AAP WF for FAM setup
+    Broker().execute(workflow='fam-test-setup', source_vm=idm_sat.name)
+
+    # Modify and copy config files to the Satellite
+    idm_fam_settings = settings.fam.server.copy()
+    # Replace hostname in the settings
+    for k, v in idm_fam_settings.items():
+        if isinstance(v, str):
+            idm_fam_settings[k] = v.replace(settings.server.hostname, idm_sat.hostname)
+    # Remove username and password as they are not used in IDM
+    idm_fam_settings['foreman_username'] = ""
+    idm_fam_settings['satellite_username'] = ""
+    idm_fam_settings['foreman_password'] = ""
+    idm_fam_settings['satellite_password'] = ""
+    # Append use_gssapi to the settings
+    idm_fam_settings['use_gssapi'] = True
```

Comment:
This is currently not consumed by any of our test playbooks in https://github.com/theforeman/foreman-ansible-modules/tree/develop/tests/test_playbooks

Should we add it to a handful/all of them?